### PR TITLE
Stop referring CLI flags in error messages if API client is unknown

### LINKED ISSUE #33138: Error messages returned by API should not refer to CLI flags/options
Some error messages returned by the daemon / API are referring to CLI flags not being valid, for example;

- [runconfig/hostconfig_windows.go#L34](https://github.com/moby/moby/blob/4af3389d43d93f50d9b4fa217de148ec45abf8cb/runconfig/hostconfig_windows.go#L34)
- [hostconfig_windows.go#L49](https://github.com/moby/moby/blob/4af3389d43d93f50d9b4fa217de148ec45abf8cb/runconfig/hostconfig_windows.go#L49)

Given that these are returned by the API, not the CLI, and we cannot assume that the _docker_ CLI is used as a client (it can be any (API) client), we should make these messages more generic and have them refer to the API-option that is not supported, not the CLI flag.

### PR DESCRIPTION
**Pull request by sbko**
Made error messages more generic. 

Fixes #33138

### CODE REVIEW

**Comment by sbko** [comment]
For some reason make unit-test didn't catch what windowsRS1 catched. I'll push a fix to tests

**Comment by fntlnz** [comment]
I would change also [api/server/router/experimental.go:14](https://github.com/moby/moby/blob/242060302a71279b7f1e9fd5031401f0aa8376ab/api/server/router/experimental.go#L14) considering the fact that also daemon options are not only be passed as parameters.

**Comment by sbko** [comment]
@fntlnz thanks! I checked only fmt.Errorf(...). I'll search for errors.New(...) as well.

File: api/server/router/build/build_routes.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 	}
 
 	if runtime.GOOS != "windows" && options.SecurityOpt != nil {
-		return nil, fmt.Errorf("the daemon on this platform does not support --security-opt to build")
+		return nil, fmt.Errorf("the daemon on this platform does not support SecurityOpt to build")
```
Comment (cpuguy83) [inline api/server/router/build/build_routes.go]: "does not support setting security options on build"

File: runconfig/hostconfig_unix.go [hunk 2]
```diff
@@ -68,11 +68,11 @@ func validateQoS(hc *container.HostConfig) error {
 	}
 
 	if hc.IOMaximumBandwidth != 0 {
-		return fmt.Errorf("invalid QoS settings: %s does not support --io-maxbandwidth", runtime.GOOS)
+		return fmt.Errorf("invalid QoS settings: %s does not support configuration of IO maximum bandwidth", runtime.GOOS)
```
Comment (cpuguy83) [inline runconfig/hostconfig_unix.go]: "does not support configuration of maximum bandwidth"

File: runconfig/hostconfig_unix.go [hunk 3]
```diff
@@ -68,11 +68,11 @@ func validateQoS(hc *container.HostConfig) error {
...
 	}
 
 	if hc.IOMaximumIOps != 0 {
-		return fmt.Errorf("invalid QoS settings: %s does not support --io-maxiops", runtime.GOOS)
+		return fmt.Errorf("invalid QoS settings: %s does not support configuration of IO maximum IOps", runtime.GOOS)
```
Comment (cpuguy83) [inline runconfig/hostconfig_unix.go]: "does not support configuration of maximum IOPs"

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

**Comment by thaJeztah** [comment]
Thanks for working on this!

File: runconfig/hostconfig_windows.go [hunk 4]
```diff
@@ -31,7 +31,7 @@ func validateNetMode(c *container.Config, hc *container.HostConfig) error {
 	}
 
 	if hc.NetworkMode.IsContainer() && hc.Isolation.IsHyperV() {
-		return fmt.Errorf("net mode --net=container:<NameOrId> unsupported for hyperv isolation")
+		return fmt.Errorf("container network stack unsupported for hyperv isolation")
```
Comment (fntlnz) [inline runconfig/hostconfig_windows.go]: I would write: Using the network stack of another  container is not supported while using Hyper-V Containers

File: api/server/router/build/build_routes.go (hunk 1 above)
Comment (fntlnz) [inline api/server/router/build/build_routes.go]: I would start with a capital letter

File: runconfig/hostconfig_unix.go (hunk 3 above)
Comment (fntlnz) [inline runconfig/hostconfig_unix.go]: capital letter here

File: runconfig/hostconfig_unix.go (hunk 2 above)
Comment (fntlnz) [inline runconfig/hostconfig_unix.go]: capital letter here

File: runconfig/hostconfig.go [hunk 5]
```diff
@@ -45,7 +45,7 @@ func validateNetContainerMode(c *container.Config, hc *container.HostConfig) err
 	parts := strings.Split(string(hc.NetworkMode), ":")
 	if parts[0] == "container" {
 		if len(parts) < 2 || parts[1] == "" {
-			return fmt.Errorf("--net: invalid net mode: invalid container format container:<name|id>")
+			return fmt.Errorf("invalid network mode: invalid container format container:<name|id>")
```
Comment (fntlnz) [inline runconfig/hostconfig.go]: capital letter here

File: runconfig/hostconfig_windows.go (hunk 4 above)
Comment (fntlnz) [inline runconfig/hostconfig_windows.go]: And also use a capital letter to start the phrase

**Review by fntlnz — CHANGES_REQUESTED** [review]

**Comment by sbko** [comment]
@fntlnz @cpuguy83 I made changes per your request. Looks like powerpc jenkins job died. Is there a way to retrigger it?

**Comment by fntlnz** [comment]
@sbko - re-triggered :innocent:

**Comment by sbko** [comment]
@fntlnz thanks! It passed as well :)

**Review by fntlnz — APPROVED** [review]

**Comment by fntlnz** [comment]
LGTM

**Comment by sbko** [comment]
@cpuguy83 I'm not sure if I need to ping you after tests are done or not. but here I am pinging you :)

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 👼

**Pull request merged**